convention.md

```diff
@@ -486,7 +494,29 @@ If a device wishes to modify any of its nodes or properties, it can
 * disconnect and reconnect with other values, or
 * set `$state=init` and then modify any of the attributes.
 
-Devices can remove old properties and nodes by publishing a zero-length payload on the respective topics.
+Devices can remove old properties and nodes by deleting the respective MQTT topics by publishing an empty message
+to those topics (an actual empty string on MQTT level, so NOT the escaped `0x00` byte, see also [empty string values](#empty-string-values)).
+
+The recommended way to add/remove child device is as follows (note: due to MQTT message ordering consistency at any stage in this process cannot be guaranteed):
+
+#### Adding:
+
+1. first publish the child-device(s), as any other device
+    1. set child-device state to `"init"`
+    1. publish child-device details (including parent details in `root` and `parent` fields)
+    1. set child-device state to `"ready"`
+1. update the parent device, as any other change
+    1. set parent state to `"init"`
+    1. update parent description (add the child ID to its `children` array)
+    1. set parent state to `"ready"`
+
+#### Removing:
+
+1. update the parent device
+    1. set parent state to `"init"`
+    1. update parent description (remove the child ID from its `children` array)
+    1. set parent state to `"ready"`
+1. clear child-device(s) topics, starting with the `$state` topic
```

Comment:
```suggestion
When adding many child devices, implementations should take care of not publishing too many parent-updates, since every controller would have to parse the description again and again.

The recommended way to add/remove child device is as follows (note: due to MQTT message ordering consistency at any stage in this process cannot be guaranteed):

#### Adding:

1. first publish any child-devices, as any other device
    1. set child-device state to `"init"`
    1. publish child-device details (including parent details in `root` and `parent` fields)
    1. set child-device state to `"ready"`
1. update the parent device, as any other change
    1. set parent state to `"init"`
    1. update parent description (add any child IDs to its `children` array)
    1. set parent state to `"ready"`

#### Removing:

1. update the parent device
    1. set parent state to `"init"`
    1. update parent description (remove any child IDs from its `children` array)
    1. set parent state to `"ready"`
1. clear any child-device(s) topics, starting with the `$state` topic
```